### PR TITLE
arangodb 3.4.1

### PR DESCRIPTION
**Pull request by fceller**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/arangodb.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  sha256 "8087d46078ecc38a9fadd1c8300e54e0a9f29ce157be1227fdc0db8533d1731d"
   head "https://github.com/arangodb/arangodb.git", :branch => "unstable"
 
   bottle do
-    sha256 "26914e81f2b90c88aa281193f50dcd26ae97ed001185d55b62af70874e11e8fa" => :mojave
-    sha256 "afff87970c8491e5dd0c76f4b6f86f6de92531332aa24cedce29a6cd8a5dcdc8" => :high_sierra
-    sha256 "5346d11d6f89c0247a066e2c277c1c15c151c5b2a66e7cf0ce564277da359b98" => :sierra
+    sha256 "a728acef3fabf95a2968d1f9fdfa8ffc6c5439c5232924f800d5d69bab42d267" => :mojave
+    sha256 "fa992f69d7cec75888924045163d57a2879c2e986611598acb78a40f61dc8d78" => :high_sierra
+    sha256 "e5c1df276d95f8fe90a96d5f34c2abb135ed98efdb2b273672fc26269ea1b07e" => :sierra
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: Don't touch bottle blocks

File: Formula/arangodb.rb [hunk 2]
```diff
@@ -49,6 +57,8 @@ def install
       system "cmake", "..", *args
       system "make", "install"
 
+      mv "#{buildpath}/usr/bin/arangodb-darwin-amd64", "#{bin}/arangodb"
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: We build from source, and do not install binaries.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/arangodb.rb (hunk 2 above)
Comment (fceller) [reply Formula/arangodb.rb]: Hi @fxcoudert, the ArangoDB Starter is a separate github project, which published a single binary used by ArangoDB.

How should this be implemented? Do I need to create another formular and require it here? Or load the source as resource and build it?

File: Formula/arangodb.rb (hunk 1 above)
Comment (fceller) [reply Formula/arangodb.rb]: Fixed, sorry for the confusion. Pick the wrong resolution when merging.

File: Formula/arangodb.rb (hunk 2 above)
Comment (fxcoudert) [reply Formula/arangodb.rb]: > load the source as resource and build it

That. But it is not clear why we want it as part of the main formula, if it is not shipped by default by upstream.

Comment (fceller) [reply Formula/arangodb.rb]: The RedHat and Debian packages contain the ArangoDB Starter as well. The load the static go binary when creating the packages.

Comment (fceller) [reply Formula/arangodb.rb]: @fxcoudert I change the formular to build the starter from scratch.

File: Formula/arangodb.rb [hunk 3]
```diff
@@ -21,16 +21,28 @@ class Arangodb < Formula
...
 
+  # see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87665
   fails_with :gcc do
     build 820
     cause "Generates incorrect code"
   end
 
+  resource "starter" do
+    url "https://github.com/arangodb-helper/arangodb.git",
+        :revision => "38953b049a38c844ffdbf667e55e60c6cfa22e70"
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: We need a comment above stating what it is. And some versioning system, we prefer (as a rule) building from archives (`
https://github.com/arangodb-helper/arangodb/archive/0.13.10.tar.gz`)

File: Formula/arangodb.rb [hunk 4]
```diff
@@ -21,16 +21,28 @@ class Arangodb < Formula
...
+  end
+
   needs :cxx11
 
   def install
     ENV.cxx11
 
+    (buildpath/"starter").install resource("starter")
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: Could we do:

```
resource("starter").stage do
  system "make", "NODOCKER=yes", "local"
  bin.install "arangodb"
end
```

Comment (fxcoudert) [reply Formula/arangodb.rb]: Also, we need to test that binary in `test do` block.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/arangodb.rb (hunk 3 above)
Comment (fceller) [reply Formula/arangodb.rb]: I took riak.rb as an example. Will change.

File: Formula/arangodb.rb (hunk 4 above)
Comment (fceller) [reply Formula/arangodb.rb]: done

File: Formula/arangodb.rb [hunk 5]
```diff
@@ -93,5 +114,8 @@ def plist; <<~EOS
     testcase = "require('@arangodb').print('it works!')"
     output = shell_output("#{bin}/arangosh --server.password \"\" --javascript.execute-string \"#{testcase}\"")
     assert_equal "it works!", output.chomp
+
+    output = shell_output("#{bin}/arangodb --version")
```
Comment (javian) [inline Formula/arangodb.rb]: Can we make it test anything else than the version that tests what the program can do other than print a number ?

Comment (fceller) [reply Formula/arangodb.rb]: I've added a start/stop test

File: Formula/arangodb.rb [hunk 6]
```diff
@@ -93,5 +116,17 @@ def plist; <<~EOS
     testcase = "require('@arangodb').print('it works!')"
     output = shell_output("#{bin}/arangosh --server.password \"\" --javascript.execute-string \"#{testcase}\"")
     assert_equal "it works!", output.chomp
+
+    start_cmd = "#{bin}/arangodb start --starter.instance-up-timeout 1m --starter.mode single"
+    ohai start_cmd
+    stdout, stderr, status = Open3.capture3(start_cmd)
```
Comment (igas) [inline Formula/arangodb.rb]: We need to have `ensure` block, so if there is something wrong we do not keep the running server on CI.

File: Formula/arangodb.rb [hunk 7]
```diff
@@ -21,16 +23,37 @@ class Arangodb < Formula
...
+
   needs :cxx11
 
   def install
     ENV.cxx11
 
+    resource("starter").stage do
+      ENV.append "GOPATH", Dir.pwd + "/.gobuild"
+      system "make", "deps"
+      system "go", "build", "-ldflags", "-X main.projectVersion=#{STARTER_VERSION} -X main.projectBuild=#{STARTER_SHA256}",
```
[the remark this answers is not in the document]
Comment (fceller) [reply Formula/arangodb.rb]: I will add the commit id for the tag.

File: Formula/arangodb.rb (hunk 6 above)
Comment (fceller) [reply Formula/arangodb.rb]: @igas what is the preferred way to kill a process in a formula?

File: Formula/arangodb.rb (hunk 7 above)
[the remark this answers is not in the document]
Comment (fceller) [reply Formula/arangodb.rb]: projectBuild must be set and cannot be undefined. The commit id for the tag 0.13.10 should be ok.

**Comment by fceller** [comment]
@fxcoudert is this now as expected?

**Comment by fceller** [comment]
@igas @javian @fxcoudert a new upstream version 3.4.2 will become available today. Shall I change this request to 3.4.2 or close and create a new one?

**Comment by fceller** [comment]
closing in favor of the upcoming 3.4.2.1

**Pull request closed without merging**